test/end-to-end/core.sh

```diff
@@ -84,193 +85,194 @@ wait_for_registry
 # services can end up on any IP.  Make sure we get the IP we need for the docker registry
 DOCKER_REGISTRY=$(oc get --output-version=v1beta3 --template="{{ .spec.portalIP }}:{{ with index .spec.ports 0 }}{{ .port }}{{ end }}" service docker-registry)
 
-registry="$(dig @${API_HOST} "docker-registry.default.svc.cluster.local." +short A | head -n 1)"
-[[ -n "${registry}" && "${registry}:5000" == "${DOCKER_REGISTRY}" ]]
+os::cmd::expect_success_and_text "dig @${API_HOST} docker-registry.default.svc.cluster.local. +short A | head -n 1 | xargs printf '%s:5000'" "${DOCKER_REGISTRY}"
 
 echo "[INFO] Verifying the docker-registry is up at ${DOCKER_REGISTRY}"
-wait_for_url_timed "http://${DOCKER_REGISTRY}/" "[INFO] Docker registry says: " $((2*TIME_MIN))
+os::cmd::try_until_success "curl --max-time 2 -fs http://${DOCKER_REGISTRY}/healthz" $((2*TIME_MIN))
```

Comment:
> Could you elaborate on what " fix the wait_for_url stuff" means?

I think we should have one helper function.  That helper should cleanly handle http and https.  Given the difficulty in setup, I want to avoid embedding the `curl` calls directly here.